app/src/lib/stores/app-store.ts

```diff
@@ -1139,6 +1141,7 @@ export class AppStore extends TypedBaseStore<IAppState> {
       return Promise.resolve(null)
     }
 
+    this.updateStashEntryCountMetric(repository)
```

Comment:
The problem you have currently is that the test repositories aren't being inserted into the `RepositoriesStore` as part of the setup, and this code path is being tripped when `AppStore._selectRepository` fires in many cases - sometimes in the test, sometimes in `AppStore.updateRepositorySelectionAfterRepositoriesChanged()` which fires after it detects the list of repositories has updated.

I'm going to spend a bit today trying to get these tests under control, but I'm curious if there's a a better place for us to do this check than "when the repository is selected".

Reply:
#7484